packages/clay-button/src/Button.tsx

```diff
@@ -85,29 +86,44 @@ const ClayButton = React.forwardRef<HTMLButtonElement, IProps>(
 			...otherProps
 		}: IProps,
 		ref
-	) => (
-		<button
-			className={classNames(className, 'btn', {
-				'alert-btn': alert,
-				'btn-block': block,
-				'btn-monospaced': monospaced,
-				'btn-outline-borderless': borderless,
-				'btn-sm': small && !size,
-				[`btn-${displayType}`]: displayType && !outline && !borderless,
-				[`btn-outline-${displayType}`]:
-					displayType && (outline || borderless),
-				'rounded-pill': rounded,
-				[`btn-${size}`]: size,
-			})}
-			ref={ref}
-			type={type}
-			{...otherProps}
-		>
-			{children}
-		</button>
-	)
+	) => {
+		const childArray = React.Children.toArray(children);
+
+		warning(
+			!(
+				childArray.length === 1 &&
+				// @ts-ignore
+				childArray[0].type?.displayName === 'ClayIcon' &&
+				typeof otherProps['aria-label'] !== 'string'
+			),
+			'Button accessibility: component has only the Icon declared, defines an `aria-label` or `title` (consult your design team) attribute that labels the interactive button that screen readers can read.'
```

Comment:
@matuzalemsteles I love this detailed warning! Small wordsmitthing suggestion:

> Button Accessibility: Component has only the Icon declared. Define an `aria-label` or `title` (consult your design team) attribute that labels the interactive button that screen readers can read.

Reply:
Thanks @ethib137, I'll update!